ts/src/htx.ts

```diff
@@ -5409,9 +5409,11 @@ export default class htx extends Exchange {
          * @returns {object} request to be sent to the exchange
          */
         const market = this.market (symbol);
+        //I'm not sure the parameters should be 'contractSize' or 'contractsize'
+        const contractSize = this.safeNumber (market, 'contractSize', 1);
         const request: Dict = {
             'contract_code': market['id'],
-            'volume': this.amountToPrecision (symbol, amount),
+            'volume': this.amountToPrecision (symbol, amount / contractSize),
```

Comment:
@EmoryLeo amountToPrecision shoul do do precisely that, it already takes into consideration the contractSize

Reply:
@EmoryLeo Are you familiar with contracts trading? When you buy a perpetual/swap (contract) you buy contracts representing the underlying asset and not the underlying asset directly, so you always have to use contract units instead of base asset. 


That's how it works in every exchange (they're only the same units base asset = contract units when the contract size is 1 otherwise you always have to do these conversions).